arch/src/x86_64/mod.rs

```diff
@@ -802,6 +802,11 @@ pub fn configure_vcpu(
     cpu_ebx |= (id as u32) << 24;
     CpuidPatch::set_cpuid_reg(&mut cpuid, 0x1, None, CpuidReg::EBX, cpu_ebx);
 
+    // Disable x2APIC support for each vcpu
+    let mut cpu_ecx = unsafe { core::arch::x86_64::__cpuid(1) }.ecx;
+    cpu_ecx &= !(1u32 << 21);
+    CpuidPatch::set_cpuid_reg(&mut cpuid, 0x1, None, CpuidReg::ECX, cpu_ecx);
+
```

Comment:
This isn't strictly necessary when AVIC is enabled (single the kernel will disable x2APIC support). But have the ability to disable x2apic (e.g. `--platform x2apic=off`) might come in handy.